lib/tools/system-calls.js

```diff
@@ -411,6 +412,21 @@ systemCallMethods.launchAVD = async function (avdName, avdArgs, language, countr
   return proc;
 };
 
+systemCallMethods.getAdbVersion = _.memoize(async function () {
+  try {
+    let adbVersion = (await this.adbExec('version'))
+      .replace(/Android\sDebug\sBridge\sversion\s([\d\.]*)[\s\w\-]*/, "$1");
+    return {
+      versionString: adbVersion,
+      versionFloat: parseFloat(adbVersion.replace(/.([^.]*)$/, '$1'))
```

Comment:
I don't think this is what we want. The point of having a `versionFloat` is to be able to quickly and easily check major-minor versions together. In my opinion this object should be return something like:
```js
let abdVersion = '1.2.3';
let parts = abdVersion.split('.');
return {
  versionString: abdVersion,
  versionFloat: parseFloat(adbVersion),
  major: parseInt(parts[0], 10),
  minor: parseInt(parts[1], 10),
  patch: parts[2] ? parseInt(parts[2], 10) : undefined,
```
Then
```js
let version = getAbdVersion();
version.versionString === '1.2.3';
version.versionFloat === 1.2;
version.major === 1;
version.minor === 2;
version.patch === 3;
```

Akin to https://github.com/appium/appium-xcode/blob/master/lib/xcode.js#L138-L154